layer/nrlf/core/tests/test_validators.py

```diff
@@ -599,6 +618,185 @@ def test_validate_category_coding_invalid_system():
+            },
+        ]
+    }
+
+    result = validator.validate(document_ref_data)
+
+    assert result.is_valid is False
+    assert result.resource.id == "Y05868-99999-99999-999999"
+    assert len(result.issues) == 1
+    assert result.issues[0].model_dump(exclude_none=True) == {
+        "severity": "error",
+        "code": "invalid",
+        "details": {
+            "coding": [
+                {
+                    "system": "https://fhir.nhs.uk/ValueSet/Spine-ErrorOrWarningCode-1",
+                    "code": "INVALID_RESOURCE",
+                    "display": "Invalid validation of resource",
+                }
+            ]
+        },
+        "diagnostics": "Invalid type coding length: 3 Type Coding must only contain a single value",
+        "expression": ["type.coding"],
+    }
+
+
+def test_validate_type_coding_invalid_system():
+    validator = DocumentReferenceValidator()
+    document_ref_data = load_document_reference_json("Y05868-736253002-Valid")
+
+    document_ref_data["type"] = {
+        "coding": [
+            {
+                "system": "http://snoooooomed/sctfffffg",
+                "code": "736253002",
+                "display": "Mental health crisis plan",
+            }
+        ]
+    }
+
+    result = validator.validate(document_ref_data)
+
+    assert result.is_valid is False
+    assert result.resource.id == "Y05868-99999-99999-999999"
+    assert len(result.issues) == 1
+    assert result.issues[0].model_dump(exclude_none=True) == {
+        "severity": "error",
+        "code": "value",
+        "details": {
+            "coding": [
+                {
+                    "system": "https://fhir.nhs.uk/ValueSet/Spine-ErrorOrWarningCode-1",
+                    "code": "INVALID_RESOURCE",
+                    "display": "Invalid validation of resource",
+                }
+            ]
+        },
+        "diagnostics": "Invalid type system: http://snoooooomed/sctfffffg Type system must be either 'http://snomed.info/sct' or 'https://nicip.nhs.uk'",
+        "expression": ["type.coding[0].system"],
+    }
+
+
+@pytest.mark.parametrize(
+    "type_str, display",
+    [
+        (type_str, display_dict["display"])
+        for type_str, display_dict in TYPE_ATTRIBUTES.items()
+    ],
+)
+def test_validate_type_coding_display_mismatch(type_str: str, display: str):
+    validator = DocumentReferenceValidator()
+    document_ref_data = load_document_reference_json("Y05868-736253002-Valid")
+    type_parts = type_str.split("|")
+    type_system = type_parts[0]
+    type_code = type_parts[1]
+
+    document_ref_data["type"] = {
+        "coding": [
+            {
+                "system": type_system,
+                "code": type_code,
+                "display": "some random display name",
+            }
+        ]
+    }
+
+    # match type to category to avoid getting that error
+    # TODO: Discuss in review if I should just ignore the error that happens to simplify test
```

Comment:
I don't think you should ignore the error personally, dont want a test having an error thats not expected

Reply:
👍  I'll remove this comment